lib/srv/server/azure_installer.go

```diff
@@ -0,0 +1,72 @@
+package server
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/Azure/azure-sdk-for-go/sdk/resourcemanager/compute/armcompute/v3"
+	"github.com/aws/aws-sdk-go/aws"
+	apievents "github.com/gravitational/teleport/api/types/events"
+	"github.com/gravitational/teleport/lib/auth"
+	"github.com/gravitational/teleport/lib/cloud/azure"
+	"github.com/gravitational/trace"
+	"golang.org/x/sync/errgroup"
+)
+
+// AzureInstallerConfig represents configuration for an Azure install script
+// executor.
+type AzureInstallerConfig struct {
+	Emitter     apievents.Emitter
+	AccessPoint auth.DiscoveryAccessPoint
+}
+
+// AzureInstaller handles running commands that install Teleport on Azure
+// virtual machines.
+type AzureInstaller struct {
+	AzureInstallerConfig
+}
+
+// NewAzureInstaller returns a new Azure installer.
+func NewAzureInstaller(cfg AzureInstallerConfig) *AzureInstaller {
+	return &AzureInstaller{
+		AzureInstallerConfig: cfg,
+	}
+}
+
+// AzureRunRequest combines parameters for running commands on a set of Azure
+// virtual machines.
+type AzureRunRequest struct {
+	Client          azure.RunCommandClient
+	Instances       []*armcompute.VirtualMachine
+	Params          []string
+	Region          string
+	ResourceGroup   string
+	ScriptName      string
+	PublicProxyAddr string
+}
+
+// Run runs a command on a set of virtual machines and then blocks until the
+// commands have completed.
+func (ai *AzureInstaller) Run(ctx context.Context, req AzureRunRequest) error {
+	g, ctx := errgroup.WithContext(ctx)
+	g.SetLimit(10)
```

Comment:
Why 10?

Reply:
I set it to the same value as in EC2 auto-join. It's arbitrary.

Reply:
That's fine. Can you just add a comment and mention the reason?

Reply:
Bump.